src/modalities/models/components/layer_norms.py

```diff
@@ -0,0 +1,92 @@
+from typing import Annotated
+
+import torch
+import torch.nn as nn
+from pydantic import BaseModel, Field
+from torch.nn import functional as F
+
+from modalities.config.lookup_enum import LookupEnum
+
+
+class LayerNormIF(nn.Module):
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        raise NotImplementedError
+
+    def __copy__(self):
+        raise NotImplementedError
+
+
+class RMSLayerNorm(LayerNormIF):
+    def __init__(self, ndim: int, epsilon: float = 1e-6):
+        """
+        Initialize the RMSNorm normalization layer.
+        Original paper: https://arxiv.org/pdf/1910.07467.pdf
+        Source code adopted from https://github.com/facebookresearch/llama/blob/a0a4da8b497c566403941ceec47c2512ecf9dd20/llama/model.py#L34C1-L77C36
+
+        Args:
+            ndim (int): The dimension of the input tensor.
+            eps (float, optional): A small value added to the denominator for numerical stability. Default is 1e-6.
```

Comment:
Epsilon is `1e-6` in the LLaMa implementation by default. However, it seems that they actually used `1e-5` themselves, see [here](https://github.com/facebookresearch/llama/issues/771#issuecomment-1761430212). `1e-5` is also the default value in PyTorch for [LayerNorm](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html) and used elsewhere for RMSNorm (e.g. [here](https://huggingface.co/mosaicml/mpt-7b/blob/039e37745f00858f0e01e988383a8c4393b1a4f5/norm.py#L28)), so it seems like a standard value that we should perhaps also use instead of `1e-6`?